karyon3-archaius2/src/main/java/com/netflix/karyon/archaius/ArchaiusProxyAutoBinder.java

```diff
@@ -3,28 +3,39 @@
 import javax.inject.Inject;
 import javax.inject.Provider;
 
-import com.google.inject.Binder;
+import com.google.inject.AbstractModule;
 import com.google.inject.Key;
+import com.google.inject.Module;
 import com.netflix.archaius.ConfigProxyFactory;
-import com.netflix.karyon.spi.AutoBinder;
+import com.netflix.archaius.annotations.Configuration;
+import com.netflix.karyon.KeyMatchers;
+import com.netflix.karyon.spi.AbstractAutoBinder;
 
 /**
  * Autobinder to create proxies for any interface containing the archaius
  * annotations
  */
-final class ArchaiusProxyAutoBinder implements AutoBinder {
+final public class ArchaiusProxyAutoBinder extends AbstractAutoBinder {
+    public ArchaiusProxyAutoBinder() {
+        super(KeyMatchers.annotatedWith(Configuration.class));
+    }
+
     @Override
-    public <T> boolean configure(Binder binder, Key<T> key) {
-        binder.bind(key).toProvider(new Provider<T>() {
-            @Inject
-            ConfigProxyFactory factory;
-            
-            @SuppressWarnings("unchecked")
+    public <T> Module getModuleForKey(final Key<T> key) {
+        return new AbstractModule() {
             @Override
-            public T get() {
-                return (T) factory.newProxy(key.getTypeLiteral().getRawType());
+            protected void configure() {
+                bind(key).toProvider(new Provider<T>() {
+                    @Inject
+                    ConfigProxyFactory factory;
```

Comment:
To mock the interface you'd bind the key directly to the mocked interface which would completely bypass the AutoBinder.  I'll update the documentation to describe this scenario.